baseapp/abci_utils.go

```diff
@@ -96,7 +97,13 @@ func ValidateVoteExtensions(
 			return fmt.Errorf("vote extensions enabled; received empty vote extension signature at height %d", currentHeight)
 		}
 
+		// Ensure that the validator has not already submitted a vote extension.
 		valConsAddr := sdk.ConsAddress(vote.Validator.Address)
+		if _, ok := cache[valConsAddr.String()]; ok {
+			continue
```

Comment:
Not sure what is preferred here (continue or throwing an error).

Reply:
I do prefer an error, if there are repeated votes then something has gone wrong imo. cc @alexanderbez

Reply:
ok cool, will push that update